drivers/volume/portworx/portworx.go

```diff
@@ -4170,6 +4178,141 @@ func (d *portworx) RecoverNode(n *node.Node) error {
 
 }
 
+// AddBlockDrives add drives to the node using PXCTL
+func (d *portworx) AddBlockDrives(n *node.Node, drivePath []string) error {
+
+	systemOpts := node.SystemctlOpts{
+		ConnectionOpts: node.ConnectionOpts{
+			Timeout:         startDriverTimeout,
+			TimeBeforeRetry: defaultRetryInterval,
+		},
+		Action: "start",
+	}
+	logrus.Infof("Getting available block drives on %s.", n.Name)
+	blockDrives, err := d.nodeDriver.GetAvailableDrives(*n, systemOpts)
+
+	if err != nil {
+		return err
+	}
+
+	eligibleDrives := make(map[string]*node.BlockDrive)
+
+	for _, drv := range blockDrives {
+		if drv.MountPoint == "" && drv.FSType == "" && drv.Type == "disk" {
```

Comment:
Also take care of PXD devices like below
`NAME="/dev/pxd!pxd746436763010209258" SIZE="6T" MOUNTPOINT="" FSTYPE="" TYPE="disk"
NAME="/dev/pxd!pxd278391229857243644" SIZE="50G" MOUNTPOINT="" FSTYPE="" TYPE="disk"`

Reply:
Fixed